### PR TITLE
Electrum

### PR DESCRIPTION
**Pull request by ghost**
@joachifm

### CODE REVIEW

**Comment by joachifm** [comment]
Cool. I'm unable to properly test this right now, but looks good to me, fwiw. Note: the travis failure is due to exceeding the log size limit.

**Comment by ehmry** [comment]
Yea, i tried to bump back the base of the branch but it still sized out.

**Comment by joachifm** [comment]
IMO this is a fairly low-risk merge anyway, as there's not much that could affect anything else here, except for the `ecdsa` bump (it and its direct revdeps build just fine for me, however).

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6305,6 +6329,25 @@ let
     };
   };
 
+
+  mnemonic =
+  let rev = "9973afdf4a9d37b7cbac4069723889eeee0e0d2a"; in
+  buildPythonPackage rec {
+    name = "mnuemonic-${builtins.substring 0 7 rev}";
```
Comment (joachifm) [inline pkgs/top-level/python-packages.nix]: Typo in `name` I think

File: pkgs/applications/misc/electrum/default.nix [hunk 2]
```diff
@@ -21,7 +22,7 @@ buildPythonPackage rec {
     requests
     slowaes
     tlslite
-  ];
+  ] ++ (if enableTrezor then [ trezor ] else [ ]);
```
Comment (joachifm) [inline pkgs/applications/misc/electrum/default.nix]: Or `optional enableTrezor trezor`

**Comment by ehmry** [comment]
@joachifm good eye, fixed those

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -5319,6 +5319,30 @@ let
...
+      sha256 = "0h093m351yzyl22zg5f4smi21d030diizc1dyq5c6ksjzchxdirl";
+    };
+    postPatch =
+      ''
+        substituteInPlace setup.py \
+            --replace /usr/include/libusb-1.0 ${pkgs.libusb1}/include/libusb-1.0
+       '';
+    meta = with stdenv.lib;
+      { description = "A Cython interface to the HIDAPI from https://github.com/signal11/hidapi";
+        license = licenses.free;
```
Comment (joachifm) [inline pkgs/top-level/python-packages.nix]: Not that it matters too much, I suppose, but hidapi is available under one of 3-clause BSD, GPL3, or their custom license.

Comment (ehmry) [inline pkgs/top-level/python-packages.nix]: Yeah, its a confusing license.

**Comment by ehmry** [comment]
Depends on #7110. @joachifm

**Comment by ehmry** [comment]
Moving to https://github.com/NixOS/nixpkgs/pull/7110

**Pull request closed without merging**